moses/ems/support/split-sentences.perl

```diff
@@ -268,7 +274,9 @@ sub preprocess {
 
 	# We stopped one token from the end to allow for easy look-ahead.
 	# Append it now.
-	$text = $text.$words[$i];
+	if (scalar(@words) > 0) {
+		$text = $text.$words[$i];
+	}
```

Comment:
It also contains a fix for warnings caused by blank (or only whitespace) lines in the input.